### ISSUE TITLE
Crash during navigation when reaching max zoom on elevation profile widget (at the bottom of the screen)

### ISSUE THREAD

**Issue by arncor** [osmandapp/OsmAnd#15135]
### Description

Crash during navigation when reaching max zoom on elevation profile widget (at the bottom of the screen)

### How to reproduce?

Start a navigation with elevation profile widget enabled. it appears at the bottom of the screen.
On this widget, zoom either by pinch out or by double tap. when max zoom is reached, another double top on the widget will cause osmand to crash

### Your Environment
OsmAnd Version: 4.2.6 Fdroid
Android/iOS version: 9
Device model: Ulefone Armor 6E

**Maps used (online or offline):**   
If you have an issue related to offline maps, tell us the exact name of the map file where the issue occurs and its edition date.

**Comment by scaidermern** [comment]
I can reproduce this on OsmAnd 4.2.7.

```
Version  OsmAnd+ 4.2.7
22.08.2022 9:17:33
Apk Version : 4.2.7 4207
Exception occured in thread Thread[main,5,main] : 
java.lang.NullPointerException: Attempt to invoke virtual method 'float com.github.mikephil.charting.highlight.Highlight.getX()' on a null object reference
	at net.osmand.plus.views.mapwidgets.widgets.ElevationProfileWidget$1.onChartSingleTapped(ElevationProfileWidget.java:244)
	at com.github.mikephil.charting.listener.BarLineChartTouchListener.onSingleTapUp(BarLineChartTouchListener.java:623)
	at android.view.GestureDetector.onTouchEvent(GestureDetector.java:730)
	at com.github.mikephil.charting.listener.BarLineChartTouchListener.onTouch(BarLineChartTouchListener.java:112)
	at com.github.mikephil.charting.charts.BarLineChartBase.onTouchEvent(BarLineChartBase.java:603)
	at android.view.View.dispatchTouchEvent(View.java:13953)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3054)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2741)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3060)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2755)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3060)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2755)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3060)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2755)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3060)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2755)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3060)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2755)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3060)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2755)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3060)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2755)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3060)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2755)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3060)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2755)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3060)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2755)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3060)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2755)
	at com.android.internal.policy.DecorView.superDispatchTouchEvent(DecorView.java:465)
	at com.android.internal.policy.PhoneWindow.superDispatchTouchEvent(PhoneWindow.java:1849)
	at android.app.Activity.dispatchTouchEvent(Activity.java:4011)
	at net.osmand.plus.activities.MapActivity.dispatchTouchEvent(MapActivity.java:1632)
	at androidx.appcompat.view.WindowCallbackWrapper.dispatchTouchEvent(WindowCallbackWrapper.java:69)
	at com.android.internal.policy.DecorView.dispatchTouchEvent(DecorView.java:423)
	at android.view.View.dispatchPointerEvent(View.java:14212)
	at android.view.ViewRootImpl$ViewPostImeInputStage.processPointerEvent(ViewRootImpl.java:5652)
	at android.view.ViewRootImpl$ViewPostImeInputStage.onProcess(ViewRootImpl.java:5455)
	at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4958)
	at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:5011)
	at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:4977)
	at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:5117)
	at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:4985)
	at android.view.ViewRootImpl$AsyncInputStage.apply(ViewRootImpl.java:5174)
	at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4958)
	at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:5011)
	at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:4977)
	at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:4985)
	at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4958)
	at android.view.ViewRootImpl.deliverInputEvent(ViewRootImpl.java:7675)
	at android.view.ViewRootImpl.doProcessInputEvents(ViewRootImpl.java:7644)
	at android.view.ViewRootImpl.enqueueInputEvent(ViewRootImpl.java:7605)
	at android.view.ViewRootImpl$WindowInputEventReceiver.onInputEvent(ViewRootImpl.java:7800)
	at android.view.InputEventReceiver.dispatchInputEvent(InputEventReceiver.java:188)
	at android.os.MessageQueue.nativePollOnce(Native Method)
	at android.os.MessageQueue.next(MessageQueue.java:336)
	at android.os.Looper.loop(Looper.java:174)
	at android.app.ActivityThread.main(ActivityThread.java:7356)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:491)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:940)

```

**Comment by Zirochkabila** [comment]
**Reproduced** OsmAnd 4.2.7, `released: 2022-07-13` and `OsmAnd~ 4.3.0#13861mqta, released: 2022-10-11`
[14 release, 15 nightly.zip](https://github.com/osmandapp/OsmAnd/files/9753458/14.release.15.nightly.zip)
